Docker integration

aoblet opened the pull request:
Warning: Must wait new tuttleofx name images on dockerhub

aoblet committed:
Fix Qt/Qml api changements with Qt5.5

aoblet committed:
Add docker files

aoblet committed:
Update readme for docker

aoblet committed:
docker: add xhost activation

aoblet committed:
Simplify Dockerfile config

- Override the mounted volume if dev mode (docker run)
- Do not use xhost (not safe)
- Use the uid and gid of the host user (need to build the image if uid
and gid are different from 1000 ...)

fabiencastan commented on Dockerfile:
```diff
@@ -0,0 +1,37 @@
+
+COPY . $BUTTLEOFX_DEV
+WORKDIR $BUTTLEOFX_DEV
+USER developer
```
We could name the user `buttle`.

aoblet replied:
Yes we could, I though about it first but I think that developer is more
accurate if the user dives inside the image. In that case the user would
dev.

No preference , as you want.
Le 28 mai 2016 13:34, "Fabien Castan" notifications@github.com a écrit :

> In Dockerfile
> https://github.com/buttleofx/ButtleOFX/pull/192#discussion_r64987849:
> 
> > -    && pip3 install numpy PyOpenGL PyOpenGL_accelerate \
> > -    && apt-get clean && apt-get autoremove && rm -rf /var/lib/apt/lists/\* /tmp/\* /var/tmp/*
> >   +
> >   +# Replace 1000 with your user / group id
> >   +RUN export uid=1000 gid=1000 && \
> > -    mkdir -p /home/developer && \
> > -    mkdir -p /etc/sudoers.d && \
> > -    echo "developer:x:${uid}:${gid}:Developer,,,:/home/developer:/bin/bash" >> /etc/passwd && \
> > -    echo "developer:x:${uid}:" >> /etc/group && \
> > -    echo "developer ALL=(ALL) NOPASSWD: ALL" > /etc/sudoers.d/developer && \
> > -    chmod 0440 /etc/sudoers.d/developer && \
> > -    chown ${uid}:${gid} -R /home/developer
> >   +
> >   +COPY . $BUTTLEOFX_DEV
> >   +WORKDIR $BUTTLEOFX_DEV
> >   +USER developer
> 
> We could name the user buttle.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/buttleofx/ButtleOFX/pull/192/files/fa1bb32b0cc5fb377648a54f33204d3f88f3ed21..15116ba4f4be461d1f36acbaaffb47fe4a033076#r64987849,
> or mute the thread
> https://github.com/notifications/unsubscribe/AG9JVoU5qqA0spGl6Krb_SB2YMylLX9Cks5qGCgugaJpZM4IouHh
> .

fabiencastan replied:
We could maybe use `archibald` to avoid confusion with the software.

aoblet committed:
Docker: user and groups management

- Remove user management from Dockerfile config
- Map host user/groups with container when runing image
- Mount /etc/passwd and /etc/group with read-only permissions (ensure
correct mapping between uid-gid and name inside container)

aoblet commented:
Ok for you ?

fabiencastan commented:
I don't know which version of docker is required.
With docker version 1.6.2, --group-add doesn't exist.

aoblet commented:
On docker 1.11.1 it works.
See [Docker PR: Adds support for specifying additional groups](https://github.com/docker/docker/pull/10717)

fabiencastan commented on buttleofx/MainWindow.qml:
```diff
@@ -79,46 +79,51 @@ ApplicationWindow {
-        if ((event.key == Qt.Key_9) && (event.modifiers & Qt.KeypadModifier)) {
-            player.changeViewer(9)
-        }
+    Item{
```
May need to add `anchors.fill : parent`

aoblet committed:
Viewer shortcuts Item: anchors fill parent

aoblet committed:
Docker: buttleofx and quickmamba to PYTHONPATH

fabiencastan commented:
Now, I get this error:

```
  File "/opt/ButtleOFX_git/buttleofx/gui/browser/browserItem.py", line 37, in BrowserItem
    thumbnailUtil = ThumbnailUtil()  # warning: use as class attribute, tuttle.core().preload(False) at init
  File "/opt/ButtleOFX_git/buttleofx/gui/browser/thumbnailUtil.py", line 12, in __init__
    tuttle.core().preload(False)
RuntimeError: boost::filesystem::status: Permission denied: "/home/developer/.tuttleofx"
```

It still looks into the internal docker user for the home folder.

fabiencastan commented:
Could be nice to add the required docker version in the readme as well as the link to [install or upgrade docker](https://docs.docker.com/engine/installation/linux).

aoblet commented:
Ok for adding docker version + link to install.
But the error: `RuntimeError: boost::filesystem::status: Permission denied: "/home/developer/.tuttleofx"` occurred because you did not re build the docker image.

aoblet committed:
Add docker version required in README

+ minor typo

fabiencastan commented:
Conflicts on:
buttleofx/MainWindow.qml

```
<<<<<<< HEAD
            Component.onCompleted: browser.fileWindow.forceActiveFocus()
=======
>>>>>>> 37d489566fb84b210bbc55554c2a775a1b7220d0
```

aoblet committed:
Active focus on browser at startup
